sdk-extensions/autoconfigure/src/main/java/io/opentelemetry/sdk/autoconfigure/internal/EnvironmentResourceProvider.java

```diff
@@ -22,7 +22,8 @@ public Resource createResource(ConfigProperties config) {
 
   @Override
   public int order() {
-    // Environment resource takes precedent over all other ResourceProviders
-    return Integer.MAX_VALUE;
+    // Environment resource takes precedent over all other ResourceProviders except
+    // ServiceInstanceIdResourceProvider.
+    return Integer.MAX_VALUE - 1;
```

Comment:
If you want you could have a package private constant in `ServiceInstanceIdResourceProvider` and reference that here:
```
return ServiceInstanceIdResourceProvider.ORDER - 1;
```